packages/core/src/shader/ShaderMacroProcessor.ts

```
@@ -0,0 +1,466 @@
+          parts.push(inst[1] as string);
+          pc++;
+          break;
+        case Op.IF_DEF:
+          pc = defines.has(inst[1] as string) || funcMacros.has(inst[1] as string) ? pc + 1 : (inst[2] as number);
+          break;
+        case Op.IF_NDEF:
+          pc = !defines.has(inst[1] as string) && !funcMacros.has(inst[1] as string) ? pc + 1 : (inst[2] as number);
+          break;
+        case Op.IF_CMP: {
+          const val = defines.get(inst[1] as string);
+          const cond = val !== undefined && ShaderMacroProcessor._evalCmpOp(Number(val) || 0, inst[2] as string, inst[3] as number);
```

ESLint

[error] Replace `·val·!==·undefined·&&` with `⏎············val·!==·undefined·&&⏎···········`

(prettier/prettier)